sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -107,6 +107,11 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     case p: Project =>
       projectToSQL(p, isDistinct = false)
 
+    case a @ Aggregate(_, _, e @ Expand(_, _, p: Project))
+      if sameOutput(e.output,
+        p.child.output ++ a.groupingExpressions.map(_.asInstanceOf[Attribute])) =>
```

Comment:
Sorry misses this: should check `isInstanceOf` before calling `asInstanceOf ` directly.
We can put all of it in one method and use it as if condition.